apps/settings/js/panels/addon_details/panel.js

```diff
@@ -0,0 +1,24 @@
+define(function(require) {
+  'use strict';
+
+  var SettingsPanel = require('modules/settings_panel');
+  var AddonDetails = require('panels/addon_details/addon_details');
+
+  return function ctor_addon_details_panel() {
+    var details;
+
+    return SettingsPanel({
+      onInit: function(panel) {
+      },
+
+      onBeforeShow: function(panel, options) {
+        details = AddonDetails(panel);
```

Comment:
It is not required to have a new instance of `AddonDetails` each time we enter the panel. We can simply create one instance of it in `onInit` and use it to do the rendering afterward.